lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -2307,6 +2307,11 @@ namespace Js
 
     BOOL JavascriptOperators::SetProperty(Var receiver, RecyclableObject* object, PropertyId propertyId, Var newValue, PropertyValueInfo * info, ScriptContext* requestContext, PropertyOperationFlags propertyOperationFlags)
     {
+        if (propertyId == PropertyIds::next && object == requestContext->GetLibrary()->GetArrayIteratorPrototype())
+        {
+            requestContext->GetLibrary()->SetArrayIteratorHasUserDefinedNext(true);
+        }
+
```

Comment:
@abchatra shall we do this for 1.2 and see if we can find a better solution later?